2-Pricing/index.html

```diff
@@ -0,0 +1,428 @@
+                                <path d="M17.9888 12.105C12.4878 9.20607 7.51029 8.41927 6.20922 10.4511C4.69349 12.819 8.74322 18.1185 15.2551 22.2867C21.767 26.455 28.2751 27.9157 29.7908 25.5478C30.8156 23.9471 29.2878 21.0529 26.1826 18.027" stroke="#EF4444" stroke-width="1.5"/>
+                            </svg>                            
+                        </div>
+                        <div>
+                            <h4 class="card-price">$159</h4>
+                        </div>
+                        <div>
+                            <p>Complete documentation</p>
+                        </div>
+
+                    </div>
+                    <div class="card-bottom">
+                        <p>2GB cloud storage</p>
+                        <button type="button" class="primary-btn button extra" data-subscription="extra">Add to cart</button>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </main>
+
+    <script>
+        if (! JSON.parse(localStorage.getItem('cart'))) {
+            localStorage.setItem('cart', '[]')
+        }
+
+        const subscriptions = {
+            bravo: {
+                name: 'Bravo',
+                monthlyPrice: 10,
+                weeklyPrice: 3,
+                subheading: 'Complete documentation',
+                description: 'Working materials in Sketch',
+                featured: false,
+            },
+            alpha: {
+                name: 'Alpha',
+                monthlyPrice: 67,
+                weeklyPrice: 23,
+                subheading: 'Working materials in EPS',
+                description: '6 months access to the library',
+                featured: true,
+            },
+            charlie: {
+                name: 'Charlie',
+                monthlyPrice: 69,
+                weeklyPrice: 25,
+                subheading: 'Working materials in PSD',
+                description: '1 year access to the library',
+                featured: false,
+            },
+            extra: {
+                name: 'Extra',
+                monthlyPrice: 159,
+                weeklyPrice: 42,
+                subheading: 'Complete documentation',
+                description: '2GB cloud storage',
+                featured: false,
+            }
+        }
+
+        const toggleView = function () {
+            const queryParams = new URLSearchParams(window.location.search)
+            const allowedViews = ['monthly', 'weekly']
+            const selectedView = queryParams.get('view')
+
+            if (! selectedView || ! allowedViews.includes(selectedView)) {
+                history.pushState(null, '', '?view=monthly')
+                return toggleView()
```

Comment:
Why the early return? Can't we just let the function keep running instead of making a recursive call?

I would try to avoid making recursive calls since it makes the code harder to understand.

Reply:
Maybe if we set `selectedView` to monthly manually